airflow/models/baseoperator.py

```diff
@@ -804,12 +804,15 @@ def render_template(      # pylint: disable=too-many-return-statements
         if not jinja_env:
             jinja_env = self.get_template_env()
 
+        from airflow.models.xcom_arg import XComArg
```

Comment:
Let's import this at the top of file. We prefere to avoid such imports because Airflow already has too much cyclic dependencies :<

Reply:
Oh, this is going to be cyclic deps as well...

Reply:
One the file is imported it doesn't cause cyclic deps, so this is fine.

https://github.com/apache/airflow/blob/master/airflow/__init__.py#L48-L56 may help as well in general terms.

Reply:
@ashb you're right that it will work. But pylint will not be happy about it once we run it over those files

Reply:
I thought it was fine with this

Reply:
Nope, pylint checks "is A imported somewhere in B and vice versa", it does not check if this import works or not.

Reply:
Hey guys,
Importing at the top of the file does cause cyclic dependencies. What is your suggested solution for moving to top of file?

Reply:
We can try to use approached mentioned by Ash. But I would leave it as it is for now